### PR TITLE
Streamline k8s deploy process

### LINKED ISSUE #3200: Implement Single-PR Kubernetes Testing/Deployment Process
## User story / feature request

As a contributor to Cal-ITP's Kubernetes infrastructure, I want to be able to test and release changes to that infrastructure (workloads, dependencies, and so on) using simple processes that execute automatically in CI/CD. [The current multi-step deployment process](https://github.com/cal-itp/data-infra/tree/main/ci) can sometimes make that a challenge.

### Acceptance Criteria

- [ ] Clean up current mismatches between deployed Kubernetes resources and the data-infra repository state
- [ ] Implement a CI/CD workflow for Kubernetes resources that takes place via a single PR - testing upon PR open or PR update (including a test deploy if a test environment exists for the resource), and deploying to prod upon merge to `main`.
- [ ] Update documentation in individual service READMEs, /ci, /docs, /.holo, and /.github to reflect changes made to the testing and deployment workflow

### Notes

The current canonical multi-step deployment process for changes to Kubernetes resources has often been bypassed for quick fixes and small changes. When the process is bypassed, it causes misalignment between some of the branches in our data-infra repository that are used by the official process and what's actually deployed in Google Cloud. As a result, the initial cleanup step is required to ensure that our repository aligns with what's deployed.

### PR DESCRIPTION
**Pull request by themightychris**
# Description

Simplifies Kubernetes review and deployment workflow by:

- Removes generating candidate branches
- Removes workflow of opening PRs against release branches to deploy Kubernetes changes
- Generating k8s diffs on any pull requests against `main` that change relevant paths
- Deploy Kubernetes changes on merge into `main`

Resolves #3200

## How has this been tested?

Manually locally

## Additional changes outside PR

- [X] Created new secret: `projects/1005246706141/secrets/jupyterhub_jupyterhub-sensitive-helm-values`
- [X] Created new secret: `projects/1005246706141/secrets/sentry_sentry-sensitive-helm-values`
- [X] Deleted unused secret: `projects/1005246706141/secrets/sentry-sensitive-yaml`
- [X] Upgraded sentry PostgreSQL database from 11 to 15

## Post-merge follow-ups

- [ ] Recreate `sentry-sentry-secret` secret from backup

### CODE REVIEW

**Comment by SorenSpicknall** [comment]
The lint task will go green once [this](https://github.com/cal-itp/data-infra/pull/3204) merges and you bring those changes into this branch.

**Review by SorenSpicknall — APPROVED** [review]
Just one small note on stewardship of a related repo.

I will likely also reopen the associated Issue or create a new documentation-specific Issue after merging to continue some docs changes related to this new workflow. For instance, the /ci README will need some updates. But those don't have to hold up merging these changes.

File: .holo/README.md
Comment (themightychris) [reply .holo/README.md:15]: Yeah I'm not sure about the best strategy there. This is a repository we actively maintain for many public/community/client consumers. Cal-ITP is pinned to a specific tagged version and it's a public open repo so it's ensured to stay static until someone bumps to a newer version. Eventually dependencies will need to be updated and at that point whoever is maintaining the infra can either fork the template or bump up to a new version. Most likely Jarvus will have already tested and documented the upgrade path to newer dependency mixes.

While there is a goal to minimize Jarvus dependencies, the project is depending on a lot of random public/community GitHub repositories for lots of things like GitHub actions and helm charts and it would be an anti-pattern to go through and fork every dependency. In cases where a Jarvus repository is linked that's just a fork of something else or a single-use repository, I'd agree with prioritizing eliminating that external dependency, but in this case this is a first-class maintained community project so I'm not sure it makes sense as a goal to just fork it into ITP preemptively

Comment (themightychris) [reply .holo/README.md:15]: @SorenSpicknall That said it would be a good issue to search for "Jarvus" throughout the codebase and make sure if any forks of community repos are being linked against, we're doing work to get any needed patches moved upstream so we can move the link back to targeting the upstream directly

**Pull request merged**
